transcendence-app/src/user/infrastructure/db/postgres/friend.postgres.repository.ts

```diff
@@ -67,20 +74,46 @@ export class FriendPostgresRepository
   async getPaginatedFriends(
     followerId: string,
     paginationDto: Required<PaginationWithSearchQueryDto>,
-  ): Promise<User[] | null> {
+  ): Promise<UserWithLevelDto[] | null> {
     const { limit, offset, sort, search } = paginationDto;
     const orderBy =
       sort === BooleanString.True ? userKeys.USERNAME : userKeys.ID;
-    const usersData = await makeQuery<User>(this.pool, {
-      text: `SELECT u.*
-      FROM ${this.table} f
-      JOIN ${table.USERS} u ON (f.${FriendKeys.FOLLOWED_ID} = u.${userKeys.ID})
-      WHERE f.${FriendKeys.FOLLOWER_ID} = $1 AND u.${userKeys.USERNAME} ILIKE $2
-      ORDER BY ${orderBy}
-      LIMIT $3
-      OFFSET $4;`,
+    const usersData = await makeQuery<UserWithLevelData>(this.pool, {
+      text: `
+          WITH ulevelwithgame AS (SELECT ul.*, g.${gameKeys.CREATED_AT}, g.${gameKeys.GAMEDURATIONINSECONDS}
+                                  FROM ${table.USER_LEVEL} ul
+                                           INNER JOIN ${table.GAME} g ON ul.${userLevelKeys.GAMEID} = g.${gameKeys.ID}
+                                  WHERE g.${gameKeys.GAMEMODE} = '${GameMode.classic}'),
```

Comment:
praise: This query works perfectly :rocket:

question: Any reason why we select the `classic` game mode here?

thought: In hindsight, I wish we have added a `"lastLevel"` column on the Users table to simplify the queries for a user with a level.

Reply:
I agree. Both here and have a last message for a chatroom. On the other hand, I've learned a lot from writing these queries.

Reply:
About classic:
Well, a user has a different level in each of the modes. It's possible that they'll have level 1 in short paddle and level 15 in classic, so if the last game they played at one point was short paddle, without this filter, the last level would be 1, and as soon as they played a classic game for an end level of 15, it would suddenly jump to 15.
We could specify what the last level is for, it just take one of them, and stay consistent.
I picked classic because I think it's our baseline.